sources/directconnect/direct-connect-gateway-attachment.go

```diff
@@ -0,0 +1,121 @@
+package directconnect
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func directConnectGatewayAttachmentOutputMapper(_ context.Context, _ *directconnect.Client, scope string, _ *directconnect.DescribeDirectConnectGatewayAttachmentsInput, output *directconnect.DescribeDirectConnectGatewayAttachmentsOutput) ([]*sdp.Item, error) {
+	items := make([]*sdp.Item, 0)
+
+	for _, attachment := range output.DirectConnectGatewayAttachments {
+		attributes, err := sources.ToAttributesCase(attachment, "tags")
+		if err != nil {
+			return nil, err
+		}
+
+		item := sdp.Item{
+			Type:            "directconnect-direct-connect-gateway-attachment",
+			UniqueAttribute: "virtualInterfaceId", // TODO: This is not correct. Added for just passing the tests validation.
```

Comment:
There is no unique attribute for this source.
For finding a unique attachment we need to pass gatewayID and virtualInterfaceID together.
What should we do here?

https://awscli.amazonaws.com/v2/documentation/api/latest/reference/directconnect/describe-direct-connect-gateway-attachments.html


I used `virtualInterfaceID` just for passing the validation in the test.

Reply:
So in SDP we *must* have a unique attribute, so what I usually do in this instance is create one that is a combination of these two. Here is an example of me doing it for another resource: https://github.com/overmindtech/aws-source/blob/65edc13c1e9722280ca6d3655760f5419ecd274b/sources/eks/addon.go#L35

Reply:
Ah I see you've already done it with `parseGatewayIDVirtualInterfaceID`. In that case all you need to do is make up a name for it and add it as an attribute. A GET requets should always return an item with the exact name that you queried for.

Reply:
Thanks for the clarification!

Addressed this with: https://github.com/overmindtech/aws-source/pull/373/commits/85107dd9eeb3a53a7665eda178aa00a5911cefe4

Reply:
Perfect